x-pack/metricbeat/module/aws/sqs/sqs_integration_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build integration
+
+package sqs
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+)
+
+func TestFetch(t *testing.T) {
+	accessKeyID, okAccessKeyID := os.LookupEnv("AWS_ACCESS_KEY_ID")
+	secretAccessKey, okSecretAccessKey := os.LookupEnv("AWS_SECRET_ACCESS_KEY")
+	sessionToken, okSessionToken := os.LookupEnv("AWS_SESSION_TOKEN")
+	defaultRegion, _ := os.LookupEnv("AWS_REGION")
+
+	if !okAccessKeyID || accessKeyID == "" {
+		t.Skip("Skipping TestFetch; $AWS_ACCESS_KEY_ID not set or set to empty")
+	} else if !okSecretAccessKey || secretAccessKey == "" {
+		t.Skip("Skipping TestFetch; $AWS_SECRET_ACCESS_KEY not set or set to empty")
+	} else {
+		tempCreds := map[string]interface{}{
+			"module":            "aws",
+			"period":            "300s",
+			"metricsets":        []string{"sqs"},
+			"access_key_id":     accessKeyID,
+			"secret_access_key": secretAccessKey,
+			"default_region":    defaultRegion,
+		}
+
+		if okSessionToken && sessionToken != "" {
+			tempCreds["session_token"] = sessionToken
+		}
+
+		sqsMetricSet := mbtest.NewReportingMetricSetV2(t, tempCreds)
+		events, err := mbtest.ReportingFetchV2(sqsMetricSet)
+		if err != nil {
+			t.Skip("Skipping TestFetch: failed to make api calls. Please check $AWS_ACCESS_KEY_ID, " +
+				"$AWS_SECRET_ACCESS_KEY and $AWS_SESSION_TOKEN in config.yml")
+		}
+
+		assert.Empty(t, err)
+		if !assert.NotEmpty(t, events) {
+			t.FailNow()
+		}
+		t.Logf("Module: %s Metricset: %s", sqsMetricSet.Module().Name(), sqsMetricSet.Name())
+
+		for _, event := range events {
+			fmt.Println("event = ", event)
+		}
+
+		errs := mbtest.WriteEventsReporterV2(sqsMetricSet, t, "/")
```

Comment:
We normally push the writing of the data into a separate test.

Reply:
Sounds good, I will split it into TestFetch and TestData. Thanks!